### ISSUE TITLE
BatchNorm Backpropagation issue with SameDiff

### ISSUE THREAD

**Issue by treo** [deeplearning4j/deeplearning4j#9029]
#### Issue Description
Originally reported in https://community.konduit.ai/t/how-to-use-sd-nn-batchnorm-in-deeplearning4j/85/4

The problem is that BatchNorm.doDiff is using BatchNormDerivative, which apparently isn't used anywhere else, and therefore doesn't get covered by the existing tests. And because of this, it got away with a broken implementation.

It _should_ be used in other places (e.g. https://github.com/eclipse/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/mkldnn/MKLDNNBatchNormHelper.java#L91-L98) but we use the DynamicCustomOp functionality there.

As most ops should be properly mapped to java by now, we should probably purge that kind of pattern for any defined op.

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: beta7

**Comment by agibsonccc** [comment]
Closing — SameDiff has been completely rewritten with a new DSP (Dynamic Shape Plan) execution engine, native plan compilation, CUDA graph replay, and Triton compilation. The architecture referenced in this issue no longer exists. If you still experience this issue with the upcoming release, please open a new issue.
